sources/packages/forms/src/form-definitions/educationprogram.json

```diff
@@ -1,5871 +1,6389 @@
+                    "calculateServer": true,
+                    "key": "cipDigit",
+                    "tags": [],
+                    "properties": {},
+                    "logic": [],
+                    "attributes": {},
+                    "overlay": {
+                        "style": "",
+                        "page": "",
+                        "left": "",
+                        "top": "",
+                        "width": "",
+                        "height": ""
+                    },
+                    "type": "hidden",
+                    "input": true,
+                    "tableView": false,
+                    "placeholder": "",
+                    "prefix": "",
+                    "suffix": "",
+                    "multiple": false,
+                    "unique": false,
+                    "hidden": false,
+                    "clearOnHide": true,
+                    "refreshOn": "",
+                    "dataGridLabel": false,
+                    "labelPosition": "top",
+                    "description": "",
+                    "errorLabel": "",
+                    "tooltip": "",
+                    "hideLabel": false,
+                    "tabindex": "",
+                    "disabled": false,
+                    "autofocus": false,
+                    "widget": {
+                        "type": "input"
+                    },
+                    "validateOn": "change",
+                    "validate": {
+                        "required": false,
+                        "custom": "",
+                        "customPrivate": false,
+                        "strictDateValidation": false,
+                        "multiple": false,
+                        "unique": false
+                    },
+                    "conditional": {
+                        "show": null,
+                        "when": null,
+                        "eq": ""
+                    },
+                    "allowCalculateOverride": false,
+                    "showCharCount": false,
+                    "showWordCount": false,
+                    "allowMultipleMasks": false,
+                    "addons": [],
+                    "inputType": "hidden",
+                    "id": "ee3hj6"
+                },
+                {
+                    "label": "SABC Code Selected",
+                    "labelWidth": "",
+                    "labelMargin": "",
+                    "customClass": "",
+                    "modalEdit": false,
+                    "defaultValue": null,
+                    "persistent": true,
+                    "protected": false,
+                    "dbIndex": false,
+                    "encrypted": false,
+                    "redrawOn": "",
+                    "customDefaultValue": "",
+                    "calculateValue": "value = data.sabcCodes.filter((d) => d.cipDigit === data.cipDigit)[0][\"sabcCode\"];",
```

Comment:
You can use array.find() instead of array.filter and avoid assigning 0 index.
![image](https://user-images.githubusercontent.com/54600590/184255369-f7f2cad2-f0a9-410e-bf6e-290af19bc6d5.png)

Reply:
Updated based on comment from @guru-aot .